fog/view/server/src/fog_view_router_server.rs

```diff
@@ -43,6 +41,33 @@ where
     report_cache_thread: Option<ReportCacheThread>,
 }
 
+/// A shard that fulfills a portion of the router's query requests.
+#[derive(Clone)]
+pub struct Shard {
+    /// The uri that this shard listens on.
+    pub uri: FogViewStoreUri,
+
+    /// The gRPC client that is used to communicate with the shard.
+    pub grpc_client: Arc<view_grpc::FogViewStoreApiClient>,
+
+    /// The `BlockRange` that this shard is responsible for providing.
+    pub block_range: BlockRange,
```

Comment:
Do these values need to be `pub`, or can they be protected behind, e.g. `block_range(&self) -> &BlockRange`-style accessors?

My default opinion on public struct members is a very defensive "only when absolutely necessary, e.g. for serialization/DMA."

Reply:
Done.

Reply:
Yep, follow-up is fine, ty